src/main/java/pl/edu/agh/firecell/engine/algorithm/TemperatureCalculation.java

```diff
@@ -0,0 +1,165 @@
+            MatterState northMatter = northCell.material().getMatterState();
+            Cell southCell = oldState.getCell(NeighbourUtils.south(cellIndex));
+            MatterState southMatter = southCell.material().getMatterState();
+
+            Cell upCell = oldState.getCell(NeighbourUtils.up(cellIndex));
+            MatterState upMatter = upCell.material().getMatterState();
+            Cell downCell = oldState.getCell(NeighbourUtils.down(cellIndex));
+            MatterState downMatter = downCell.material().getMatterState();
+
+            Cell eastCell = oldState.getCell(NeighbourUtils.east(cellIndex));
+            MatterState eastMatter = eastCell.material().getMatterState();
+            Cell westCell = oldState.getCell(NeighbourUtils.west(cellIndex));
+            MatterState westMatter = westCell.material().getMatterState();
+
+            if (northMatter == MatterState.SOLID && southMatter == MatterState.SOLID)
+                yTemp = computeConductivity(northCell, oldCell, southCell);
+            else if (northMatter == MatterState.SOLID)
+                yTemp = computeConductivity(northCell, oldCell, new Cell(oldCell));
+            else if (southMatter == MatterState.SOLID)
+                yTemp = computeConductivity(new Cell(oldCell), oldCell, southCell);
+
+            if (upMatter == MatterState.SOLID && downMatter == MatterState.SOLID)
+                zTemp = computeConductivity(upCell, oldCell, downCell);
+            else if (upMatter == MatterState.SOLID)
+                zTemp = computeConductivity(upCell, oldCell, new Cell(oldCell));
+            else if (downMatter == MatterState.SOLID)
+                zTemp = computeConductivity(new Cell(oldCell), oldCell, downCell);
+
+            if (eastMatter == MatterState.SOLID && westMatter == MatterState.SOLID)
+                xTemp = computeConductivity(eastCell, oldCell, westCell);
+            else if (eastMatter == MatterState.SOLID)
+                xTemp = computeConductivity(eastCell, oldCell, new Cell(oldCell));
+            else if (westMatter == MatterState.SOLID)
+                xTemp = computeConductivity(new Cell(oldCell), oldCell, westCell);
+
+        } catch (IndexOutOfBoundsException ignored) {
+            if (oldState.hasCell(NeighbourUtils.up(cellIndex))){
+                zTemp = computeConductivity(oldState.getCell(NeighbourUtils.up(cellIndex)), oldCell, new Cell(oldCell));
+            }
+        }
+        return yTemp + zTemp + xTemp;
+    }
+
+    private double computeConductivity(Cell former, Cell middle, Cell latter) {
+        return - CONDUCTIVITY_COEFFICIENT * (tempDiff(middle, former) + tempDiff(middle, latter));
+    }
+
+    private static double tempDiffAbs(Cell cellOne, Cell cellTwo) {
+        return Math.abs(tempDiff(cellOne, cellTwo));
+    }
+
+    private static double tempDiff(Cell cellOne, Cell cellTwo) {
+        return cellOne.temperature() - cellTwo.temperature();
+    }
```

Comment:
I would not agree, the aim of this function is to point diffrence between temperature in two given cells. For me it is more readable then longer form like cell1.temperature() - cell2.temperature(). It is more abstract. Maybe it can be named in other way for example twoCellsTemperatureDiffrence().